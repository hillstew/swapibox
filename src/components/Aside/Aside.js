import React from "react";
import PropTypes from "prop-types";

export const Aside = props => {
  const { openingCrawl, title, year } = props.film;
  return (
    <aside className="aside">
      <p className="scrolling-text">{openingCrawl}</p>
      <p className="scrolling-text">{title}</p>
      <p className="scrolling-text">{year}</p>
    </aside>
  );
};

Aside.propTypes = {
  film: PropTypes.object
}
