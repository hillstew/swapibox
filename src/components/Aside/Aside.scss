@font-face {
  font-family: "Galaxy";
  src: url(./fonts/SFdistantGalaxy.ttf);
}

.aside {
  background-color: rgb(0, 0, 29);
  color: #80808079;
  font-family: "Galaxy";
  font-size: 2rem;
  overflow: scroll;
  padding: 0 20px;
  position: relative;
  text-align: left;
  
  .scrolling-text {
    animation: scrolling 30s linear 1;
    letter-spacing: 1px;
    line-height: 2.5rem;
    margin-top: 0;
    position: relative;
    top: 1em;
  }
}

@keyframes scrolling {
  0% {
    top: 20em;
  }
  100% {
    top: -40em;
  }
}
