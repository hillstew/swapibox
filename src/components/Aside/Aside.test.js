import React from "react";
import { Aside } from "../Aside/Aside";
import { shallow } from "enzyme";

const filmMock = {
  openingCrawl: "It is a dark time for the",
  title: "The Empire Strikes Back",
  year: "1980-05-17"
};

describe("Aside", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Aside film={filmMock} />);
    expect(wrapper).toMatchSnapshot();
  });
});
