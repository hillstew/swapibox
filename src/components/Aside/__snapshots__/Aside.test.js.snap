// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aside should match snapshot 1`] = `
<aside
  className="aside"
>
  <p
    className="scrolling-text"
  >
    It is a dark time for the
  </p>
  <p
    className="scrolling-text"
  >
    The Empire Strikes Back
  </p>
  <p
    className="scrolling-text"
  >
    1980-05-17
  </p>
</aside>
`;
